rholang/src/main/scala/coop/rchain/rholang/interpreter/compiler/normalize.scala

```diff
@@ -722,17 +734,52 @@ object ProcNormalizeMatcher {
         } yield ProcVisitOutputs(
           input.par.prepend(
             Send(
-              nameMatchResult.chan,
+              nameMatchResult.par,
               dataResults._1,
               persistent,
               ParLocallyFree
-                .locallyFree(nameMatchResult.chan, input.env.depth) | dataResults._3,
-              ParLocallyFree.connectiveUsed(nameMatchResult.chan) || dataResults._4
+                .locallyFree(nameMatchResult.par, input.boundMapChain.depth) | dataResults._3,
+              ParLocallyFree.connectiveUsed(nameMatchResult.par) || dataResults._4
             )
           ),
-          dataResults._2.knownFree
+          dataResults._2.freeMap
         )
 
+      case p: PSendSynch =>
+        val identifier = UUID.randomUUID().toString
+        val nameVar    = new NameVar(identifier)
+
+        val send: PSend = {
+          p.listproc_.prepend(new PEval(nameVar))
+          new PSend(p.name_, new SendSingle(), p.listproc_)
+        }
+
+        val receive: PInput = {
+
+          val listName = new ListName()
+          listName.add(new NameWildcard)
```

Comment:
This can only bind to `NameWildcard` which means the continuation can use the return value. 
This results in that only `ack` situation can be used in this syntax while return value type is not applicable, is that correct?
I mean a function like https://github.com/rchain/rchain/blob/dev/casper/src/main/resources/ListOps.rho#L45-L48 can not used in this situation.

Reply:
Oh. Actually the https://github.com/rchain/rchain/blob/dev/casper/src/main/resources/ListOps.rho#L45-L48 can be solved by complexConsume, right?

Reply:
Yes, it can be simplified with a complex source, assuming the return channel convention is followed.